test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs

```diff
@@ -45,10 +45,25 @@ public void CPPRunAllTestExecutionPlatformx64(RunnerInfo runnerInfo)
 
         [TestMethod]
         [NetFullTargetFrameworkDataSource]
-        public void WebTestRunAllTests(RunnerInfo runnerInfo)
+        public void WebTestRunAllTestsWithRunSettings(RunnerInfo runnerInfo)
         {
             AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerInfo);
-            WebTestRunAllTests(runnerInfo.RunnerFramework);
+            var runSettingsFilePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid() + ".runsettings");
+
+            //test the simulateThinkTimes setting for WebTestRunConfiguration in run settings
+            var runSettingsXml = $@"<?xml version='1.0' encoding='utf-8'?>
+                                <RunSettings>
+                                    <WebTestRunConfiguration simulateThinkTimes='true' />
+                                    <RunConfiguration>
+                                        <TargetFrameworkVersion>{FrameworkArgValue}</TargetFrameworkVersion>
+                                    </RunConfiguration>
+                                </RunSettings>";
+
+            IntegrationTestBase.CreateRunSettingsFile(runSettingsFilePath, runSettingsXml);
+
+            //minExecutionTimeInSeconds is set to 30 here as the web test has a request with think time equal to 30 seconds
+            //therefore, the test will at least run for 30 seconds
+            WebTestRunAllTests(runnerInfo.RunnerFramework, runSettingsFilePath, 30);
```

Comment:
Can we reduce this please.
This will simply add 30 sec to the overall run

Reply:
I discussed this at length with Satya. It is added for a purpose. Also, we are replacing the old test with this one so gaining few seconds from the old test.

Reply:
Using iterationCount/web test result file size to validate the web test config.
Removed this think time logic.